### PR TITLE
Make job state param optional again

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes https://github.com/galaxyproject/galaxy/pull/16778/files/4c4f949c0ed752656fb567ad0cde74efa51cc015#r1403685512

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/schema/jobs.py [hunk 1]
```diff
@@ -110,8 +110,8 @@ class SearchJobsPayload(Model):
         title="Inputs",
         description="The inputs of the job.",
     )
-    state: JobState = Field(
-        default=Required,
+    state: Optional[JobState] = Field(
+        default=JobState.OK,
```
Comment (nsoranzo) [inline lib/galaxy/schema/jobs.py]: If I am not mistaken, the previous default was `None`, which resulted in searching jobs in the Job.states.NEW, Job.states.QUEUED, Job.states.WAITING, Job.states.RUNNING, Job.states.OK states.

**Review by jmchilton — APPROVED** [review]

File: lib/galaxy/schema/jobs.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/schema/jobs.py]: This was marked as resolved, but I don't think it has been?

Comment (mvdbeek) [reply lib/galaxy/schema/jobs.py]: eh, forgot to push

**Review by nsoranzo — APPROVED** [review]
Thanks!

**Pull request merged**